test/UnitInvalidation.cpp

```diff
@@ -0,0 +1,187 @@
+        renderArea(windowTwo, 0);
+
+        LOG_TST("Tab switch and edit: window one");
+
+        // Switch to sheet two in window one
+        helpers::sendTextFrame(windowOne, "setclientpart part=1", testname);
+        setupSession(windowOne); // for good measure
+
+        // Get some tiles here too for good measure
+        renderArea(windowOne, 1);
+
+        helpers::sendTextFrame(windowOne, "mouse type=move x=1830 y=475 count=1 buttons=0 modifier=0", testname);
+        helpers::sendTextFrame(windowOne, "mouse type=buttondown x=1830 y=475 count=1 buttons=1 modifier=0", testname);
+        helpers::sendTextFrame(windowOne, "mouse type=buttonup   x=1830 y=475 count=1 buttons=1 modifier=0", testname);
+        helpers::sendTextFrame(windowOne, "mouse type=buttondown x=1830 y=475 count=2 buttons=1 modifier=0", testname);
+        helpers::sendTextFrame(windowOne, "mouse type=buttonup   x=1830 y=475 count=2 buttons=1 modifier=0", testname);
+
+        std::string message;
+
+        // Select cell B2 and start editing
+        message = helpers::getResponseString(windowOne, "celladdress:", testname);
+        LOK_ASSERT_EQUAL(message, std::string("celladdress: B2"));
+
+        message = helpers::getResponseString(windowOne, "cursorvisible:", testname);
+        LOK_ASSERT_EQUAL(message, std::string("cursorvisible: true"));
+
+        // Enter a number of '1's + enter
+        helpers::sendTextFrame(windowOne, "textinput id=0 text=11111111", testname);
+        helpers::sendTextFrame(windowOne, "key type=input char=13 key=1280", testname);
+        helpers::sendTextFrame(windowOne, "key type=up char=0 key=1280", testname);
+
+        LOG_TST("Wait for window one invalidations");
+        std::vector<std::string> invalidates;
+
+        invalidates = helpers::getAllResponsesTimed(windowOne, "invalidatetiles:", testname, std::chrono::seconds(2));
+        LOK_ASSERT(invalidates.size() > 0);
+
+        bool invalidated = false;
+        for (auto &i : invalidates)
+        {
+            if (i.starts_with("invalidatetiles: part=1"))
+            {
+                invalidated = true;
+                break;
+            }
+        }
+        LOK_ASSERT(invalidated);
+        // We expect invalidates on both part 1 (current) and 0 (dependent)
+
+        LOG_TST("Wait for window two invalidations");
+
+        // Notice no invalidation on window two
+        invalidates = helpers::getAllResponsesTimed(windowTwo, "invalidatetiles:", testname, std::chrono::seconds(2));
+
+        // We expect to get an invalidate for our current view, due to the dependency updating.
+        LOK_ASSERT(invalidates.size() > 0);
+        invalidated = false;
+        for (auto &i : invalidates)
+        {
+            if (i.starts_with("invalidatetiles: part=0"))
+            {
+                invalidated = true;
+                break;
+            }
+        }
+        LOK_ASSERT(invalidated);
```

Comment:
Once we filter for part 0 invalidations in the windowTwo messages, this fails without the fix, I think that was the only missing piece here.

Reply:
Makes sense - sorry - didn't get fully to the end of this but ... ;-) got somewhere at least.